pkg/reconciler/buildrun/resources/sources/git.go

```diff
@@ -0,0 +1,76 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package sources
+
+import (
+	"fmt"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/build/pkg/config"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// AppendGitStep appends the Git step and results and volume if needed to the TaskSpec
+func AppendGitStep(
+	cfg *config.Config,
+	taskSpec *tektonv1beta1.TaskSpec,
+	source buildv1alpha1.Source,
+	name string,
+) {
+	// append the result
+	taskSpec.Results = append(taskSpec.Results, tektonv1beta1.TaskResult{
```

Comment:
would it be possible to collocate all Tekton Results definition in the `taskrun.go` file, together with the digest and size. It helps to easily find where all results are defined. I also see this file only populating them, and the `taskrun.go` one defining them.

Reply:
Conceptionally that feels incorrect because then taskrun.go would need to know which results the different source types create. git creates a `commit-sha` today, eventually we might add for example a `commit-author`. HTTP might eventually create a `size` result. An eventual SVN source would have a `revision` result.